docs/docs/dev_docs/contracts/syntax/functions.md

```diff
@@ -2,53 +2,233 @@
+
+`Aztec.nr` has a module dedicated to its oracles. If you are interested, you can view them by following the link below:
+#include_code oracles-module /yarn-project/aztec-nr/aztec/src/oracle.nr rust
+
+### A few useful inbuilt oracles
+
+- [`compute_selector`](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/aztec/src/oracle/compute_selector.nr) - Computes the selector of a function. This is useful for when you want to call a function from within a circuit, but don't got an interface at hand and don't want to hardcode the selector in hex.
+- [`debug_log`](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/aztec/src/oracle/debug_log.nr) - Provides a couple of debug functions that can be used to log information to the console. 
+- [`auth_witness`](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/aztec/src/oracle/auth_witness.nr) - Provides a way to fetch the authentication witness for a given address. This is useful when building account contracts to support approve-like functionality.
+- [`get_l1_to_l2_message`](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/aztec/src/oracle/get_l1_to_l2_message.nr) - Useful for application that receive messages from L1 to be consumed on L2, such as token bridges or other cross-chain applications.
+- [`notes`](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/aztec/src/oracle/notes.nr) - Provides a lot of functions related to notes, such as fetches notes from storage etc, used behind the scenes for value notes and other pre-build note implementations.
+
+
+---
+
+## Calling functions from other functions
+
+### Private -> Private
+
+In Aztec Private to Private function calls are handled by the [private kernel circuit](../../../concepts/advanced/circuits/kernels/private_kernel.md), and take place on the user's device.
+Behind the scenes, the `Aztec RPC Server` (the beating heart of Aztec that runs in your wallet) will execute all of the functions in the desired order "simulating" them in sequence. For example, a very common use-case of Private to Private interaction is calling another private function from an `account contract` (Account contracts are a general concept, more information about them can be found [here](../../wallets/writing_an_account_contract.md)).
+
+Take, for example, the following call stack:
+```
+AccountContract::entrypoint
+    |-> Foo::example_call
+        | -> Bar::nested_call
+    |-> Baz::example_call
+```
+
+In the example above the Account Contract has been instructed to call two external functions. In the first function all, to `Foo::example_call` a further nested call is performed to `Bar::nested_call`. Finally the account contract makes one last call to `Baz::example_call`.
+
+Lets further illustrate what these examples could look like
+<!-- TODO: should these move into the docs examples -->
```

Comment:
Is this to be done before merging this PR?

Reply:
No. The example is just to show the ordering, can stand as is.